packages/react-app/src/components/giveaways/cards/Card.tsx

```diff
@@ -10,13 +10,20 @@ import { GetParticipants } from '../../../lib/queryClient';
 import { Giveaway, ParticipationState } from '../../../lib/types';
 import ParticipationService from '../../../services/giveawayparticipation';
 import ParticipationButton from '../participation/ParticipationButton';
+import PendingApprovalBanner from '../participation/PendingApprovalBanner';
 
-const GiveawayCard = (giveaway: Giveaway) => {
+type GiveawayCardProps = {
+  giveaway: Giveaway;
+  onWinnersGeneration: () => void;
+};
+
+const GiveawayCard = ({ giveaway, onWinnersGeneration }: GiveawayCardProps) => {
   const navigate = useNavigate();
   const userInfo = useUserInfo();
   const { data: participants } = GetParticipants(giveaway._id);
+  const [participation, setParticipation] = useState<ParticipationState>();
   const isWinner = ParticipationService.wonGiveaway(giveaway, userInfo);
-  const [isAllowed, setIsAllowed] = useState(true);
+  const nrPending = giveaway.nrPendingParticipants;
```

Comment:
This is totally optional, just personal preference, but it would be great if we did not rename variables so we don't lose track of what we are working with
```suggestion
  const { nrPendingParticipants } = giveaway;
```